src/main/java/org/fogbowcloud/saps/engine/core/catalog/Catalog.java

```diff
@@ -0,0 +1,49 @@
+package org.fogbowcloud.saps.engine.core.catalog;
+
+import java.sql.SQLException;
+import java.sql.Timestamp;
+import java.util.Date;
+import java.util.List;
+
+import org.fogbowcloud.saps.engine.core.catalog.exceptions.CatalogException;
+import org.fogbowcloud.saps.engine.core.catalog.exceptions.TaskNotFoundException;
+import org.fogbowcloud.saps.engine.core.catalog.exceptions.UserNotFoundException;
+import org.fogbowcloud.saps.engine.core.dispatcher.notifier.Ward;
+import org.fogbowcloud.saps.engine.core.model.SapsImage;
+import org.fogbowcloud.saps.engine.core.model.SapsUser;
+import org.fogbowcloud.saps.engine.core.model.enums.ImageTaskState;
+
+public interface Catalog {
+
+    SapsImage addTask(String taskId, String dataset, String region, Date date, int priority, String user,
```

Comment:
this addTask method is complex. please, add a doc to it. I want to one what is the meaning of each variable. Also, tell when does the exception is thrown.

Reply:
Doc added